File: queue-frontend/src/app/dtos/users/ResponseUserInfoDto.ts
export interface ResponseUserInfoDto {

  userId: string,
  username: string,
  name: string,
  surname: string,
  phone: string,
  email: string,
  role: string,
  counterNumber: number,
  active: boolean,
  createdAt: string,
  updatedAt: string,
  serviceNames: string[]
}

File: queue/src/main/java/br/com/queue/dtos/user/get_user/ResponseUserInfoDto.java
package br.com.queue.dtos.user.get_user;

import java.util.Set;

public record ResponseUserInfoDto(
        String userId,
        String username,
        String name,
        String surname,
        String phone,
        String email,
        String role,
        Integer counterNumber,
        Boolean active,
        String createdAt,
        String updatedAt,
        Set<String> serviceNames
) {
}
